### PR TITLE
Add operator deployment readiness probe

### LINKED ISSUE #1871: Add operator deployment readiness probes
Please, answer some short questions which should help us to understand your problem / question better?

- **Which image of the operator are you using?** registry.opensource.zalan.do/acid/postgres-operator:v1.8.0
- **Where do you run it - cloud or metal? Kubernetes or OpenShift?** [AWS K8s | GCP ... | Bare Metal K8s] kind
- **Are you running Postgres Operator in production?** [yes | no] no
- **Type of issue?** [Bug report, question, feature request, etc.] feature request

It would be nice to implement internally and create an embedded REST endpoints for operator readiness probe. Otherwise, there is no possible to analyse if operator correctly starting or not based on Deployment state.

### PR DESCRIPTION
**Pull request by dmvolod**
Fixes #1871

### CODE REVIEW

**Comment by FxKu** [comment]
At Zalando we consider [liveness probes](https://srcco.de/posts/kubernetes-liveness-probes-are-dangerous.html)) as something to avoid. The readiness probe might a good addition, though.

**Comment by dmvolod** [comment]
> At Zalando we consider [liveness probes](https://srcco.de/posts/kubernetes-liveness-probes-are-dangerous.html) as something to avoid. The readiness probe might a good addition, though.

Good point, thanks. Will adopt ti utilize readiness probe only.

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -455,13 +455,14 @@ func (c *Controller) Run(stopCh <-chan struct{}, wg *sync.WaitGroup) {
...
 	go c.runPostgresqlInformer(stopCh, wg)
 	go c.clusterResync(stopCh, wg)
-	go c.apiserver.Run(stopCh, wg)
 	go c.kubeNodesInformer(stopCh, wg)
 
 	if c.opConfig.EnablePostgresTeamCRD {
 		go c.runPostgresTeamInformer(stopCh, wg)
 	}
 
+	go c.apiserver.Run(stopCh, wg)
```
[the remark this answers is not in the document]
Comment (dmvolod) [reply pkg/controller/controller.go:464]: Oh, seems to we don't need to modify this sequence, as API server should be running as latest go-routine,  but it's not implemented. Changes reverted

**Comment by FxKu** [comment]
:+1:

**Comment by idanovinda** [comment]
👍

**Pull request merged**